src/AzureSqlActionHelper.ts

```diff
@@ -146,16 +166,50 @@ export default class AzureSqlActionHelper {
         else if (maximumVersion === sqlPackagePatInstalledWithSSDT[1]) {
             sqlPackagePath = sqlPackagePatInstalledWithSSDT[0];
         }
+        
 
-        if (!sqlPackagePath) {
-            throw new Error('Unable to find the location of SqlPackage.exe');
+        if (sqlPackagePath == '') {
+            throw new Error('Unable to find the location of SqlPackage');
         }
 
-        core.debug(`SqlPackage.exe found at location: ${sqlPackagePath}`);
+        core.debug(`SqlPackage ${maximumVersion} selected at location: ${sqlPackagePath}`);
         return sqlPackagePath;
     }
 
-    private static async _getSqlPackageInstalledWithSSDT(): Promise<[string, number]> {
+    /** SqlPackage returns a multi-part version number major.minor.patch
+     * sqlpackage doesn't append -preview to the version number, but if added in the future, this method will handle it
+     * This method returns the version as a SemVer object for comparison to find the highest version
+     */
+    private static async _getSqlPackageExecutableVersion(sqlPackagePath: string): Promise<semver.SemVer> {
+        let versionOutput = '';
+        await exec.exec(`"${sqlPackagePath}"`, ['/version'], {
+            listeners: {
+                stdout: (data: Buffer) => versionOutput += data.toString()
+            }
+        });
+
+        let version = semver.coerce(versionOutput.trim());
+        if (!semver.valid(version) || version === null) {
+            core.debug(`Unable to parse version ${versionOutput} of SqlPackage at location ${sqlPackagePath}`);
+            return new semver.SemVer('0.0.0');
+        }
+
+        return version;
+    }
+
+    private static async _getSqlPackageExeInstalledDotnetTool(): Promise<[string, semver.SemVer]> {
+        let globalDotnetToolsPath = path.join(process.env['USERPROFILE'] as string, '.dotnet', 'tools');
+        let sqlPackagePath = path.join(globalDotnetToolsPath, 'SqlPackage.exe');
+        if (fs.existsSync(sqlPackagePath)) {
+            core.debug(`SqlPackage (installed with dotnet tool) found at location: ${sqlPackagePath}`);
```

Comment:
Are there going to be double debug statements (due to debug statements also existing in _getSqlPackageExecutablePath)

Reply:
yep, that would have been too noisy - thanks - cleaned up and moved all positive case debug messages to the specific functions